tests/python/frontend/tflite/test_forward.py

```diff
@@ -868,74 +869,177 @@ def test_forward_l2_pool2d():
 # -----------
 
 
-def _test_tflite2_quantized_convolution(
-    input_shape, kernel_shape, dilations, strides, padding, data_format
+def _test_tflite2_quantized_conv2d(
+    input_shape,
+    weights_shape,
+    dilations,
+    strides,
+    padding,
+    dtype="int8",
+    quantize_per_channel=False,
 ):
     """One iteration of TFLite2 quantized convolution with given shapes and attributes"""
-    data_format = "channels_last" if "NHWC" else "channels_first"
-    data = np.random.uniform(0, 1, input_shape).astype("float32")
-    kernel = np.random.uniform(0, 1, kernel_shape).astype("float32")
 
-    data_in = tf.keras.layers.Input(shape=data.shape[1:])
-    conv = tf.keras.layers.Conv2D(
-        filters=kernel_shape[3],
-        kernel_size=(kernel_shape[0], kernel_shape[1]),
-        strides=strides,
-        padding=padding,
-        data_format=data_format,
-        activation="relu",
-        use_bias=False,
-    )(data_in)
-    keras_model = tf.keras.models.Model(data_in, conv)
-    keras_model.layers[1].set_weights([kernel])
+    dtype_min, dtype_max = test_tflite.get_range_for_dtype_str(dtype)
+    channels = weights_shape[0]
 
-    # To create quantized values with dynamic range of activations, needs representative dataset
-    def representative_data_gen():
-        for i in range(1):
-            yield [data]
+    input_scale = np.random.random() * 0.1
+    input_zp = np.random.randint(dtype_min, dtype_max)
+    in_tensor = test_tflite.Tensor(
+        data_type=dtype,
+        shape=input_shape,
+        quantization=test_tflite.Quantization(scale=[input_scale], zero_point=[input_zp]),
+    )
 
-    tflite_model_quant = _quantize_keras_model(keras_model, representative_data_gen)
+    # Weights in TFLite 2 are symmetric, i.e the zero point is at 0
+    if quantize_per_channel:
+        weights_scale = [np.random.random() * 0.1 for i in range(channels)]
+        weights_zp = [0 for i in range(channels)]
+    else:
+        weights_scale = [np.random.random() * 0.1]
+        weights_zp = [0]
+    weights_quantization = test_tflite.Quantization(
```

Comment:
In general I'm a bit concerned by how verbose you need to be to create a single operator - especially vs. the code we're replacing here. Can we think of any ways to reduce the amount of boilerplate required here to create the Tensors?

Reply:
Some ways we could reduce the amount of code that I can come up with:
(1) Use the same function for both, conv2d and depthwise2d and have a parameter to distinguish between the two cases, to make sure we set the quantized dimension etc correctly. 
(2) Since TFLite2 is focusing on int8, we could make it a default data type for tensors, in most cases that would mean that there will be one parameter less to set when creating a tensor.
(3) We can give the operator classes the power to create tensors themselves that make sense for the specific operator, e.g.
conv2d = test_tflite.Conv2DOperator(...)
weights_tensor = conv2d.make_weights(shape, <some optional arguments>)
We can have default data type, qnn params that make sense and optionally fill it with data. Going further, if we tag a tensor with whether it is an input or output, we could save tensors into the op object itself and extract them during json creation. Then we'd essentially only need to pass conv2d to the generate_tflite_model. We might want to think whether it is a good idea to introduce that kind of coupling though. 

In general, I'm note sure you can create a tensor directly without specifying its qnn params, shape and dtype, but maybe you have some more ideas?

Reply:
I like the idea of the operators being able to create their own weights/biases. That is somewhat analogous to what happens when you create the operators with Keras/TF. This way if we need to create other Conv2D tests, we don't need to replicate a lot of boilerplate. I'm also inclined to agree that we can default the data type to int8.